packages/roosterjs-editor-core/lib/coreApi/selectTable.ts

```diff
@@ -0,0 +1,206 @@
+
+    let css = '';
+
+    let isFirst = true;
+    for (let i = tr1; i <= tr2; i++) {
+        firstSelected = null;
+        lastSelected = null;
+
+        for (let j = td1; j <= td2; j++) {
+            if (isFirst) {
+                isFirst = false;
+            } else {
+                css += ',';
+            }
+
+            let selector =
+                contentDivSelector +
+                ' #' +
+                table.id +
+                tbodySelector +
+                ' >tr:nth-child(' +
+                i +
+                ')>td:nth-child(' +
+                j +
+                ')';
+            if (table.querySelector(selector)) {
+                css += selector;
+            } else {
+                selector = selector.replace('td', 'th');
+                if (table.querySelector(selector)) {
+                    css += selector;
+                }
+            }
+        }
+
+        const rowSelector = 'tr:nth-child(' + i + ')>';
+        const firstSelector = `${rowSelector}td:nth-child(${td1}),${rowSelector}th:nth-child(${td1})`;
+        const lastSelector = `${rowSelector}td:nth-child(${td2}),${rowSelector}th:nth-child(${td2})`;
+
+        firstSelected = table.querySelector(firstSelector);
+        lastSelected = table.querySelector(lastSelector)!;
+        if (firstSelected && lastSelected) {
+            const rowRange = new Range();
+            rowRange.setStartBefore(firstSelected);
+            rowRange.setEndAfter(lastSelected);
+            ranges.push(rowRange);
+        }
+    }
+
+    css += '{background-color: rgba(198,198,198,0.7) !important;}';
+
+    return { css, ranges };
+}
+
+function select(table: HTMLTableElement, coordinates: TableSelection, core: EditorCore): Range[] {
+    const doc = core.contentDiv.ownerDocument;
+    const contentDivId = core.contentDiv.id;
+
+    const contentDivSelector = '#' + contentDivId;
+    let { css, ranges } = buildCss(table, coordinates, contentDivSelector);
+    if (doc) {
+        unselect();
+        if (styleElement && styleElement.ownerDocument != doc) {
+            styleElement.parentElement.removeChild(styleElement);
+            styleElement = null;
+        }
+        if (!styleElement) {
+            styleElement = doc.createElement('style');
+            doc.head.appendChild(styleElement);
+        }
+        styleElement.sheet.insertRule(css);
+    }
+
+    return ranges;
+}
+
+function unselect() {
+    if (styleElement?.sheet?.cssRules) {
+        while (styleElement.sheet.cssRules.length > 0) {
+            styleElement.sheet.deleteRule(0);
+        }
+    }
+}
+
+/**
+ * Make the first Cell of a table selection always be on top of the last cell.
+ * @param input Table selection
+ * @returns Table Selection where the first cell is always going to be first selected in the table
+ * and the last cell always going to be last selected in the table.
+ */
+function normalizeTableSelection(input: TableSelection): TableSelection {
```

Comment:
I think we do that in tableCellSelection plugin, but might as well add it here since this is going to be a api that is going to be usable from all the Editor.

I will add this.